File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSum = nums.front(), sum = nums.front();

        for(int i = 1; i < nums.size(); i++){
            

            if(sum <= 0)
                sum = 0;
            sum += nums[i];
            maxSum = max(maxSum, sum);
        }
        return maxSum;
    }
};

File: 53-maximum-subarray/maximum-subarray.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
        int maxSum = int.MinValue, sum = 0;

        for(int i = 0; i < nums.Length; i++){
            sum += nums[i];
            maxSum = Math.Max(maxSum, sum);
            if(sum < 0) sum = 0;
        }
        return maxSum;
    }
}